diff --git a/node_modules/react-native-code-highlighter/dist/commonjs/lib/CodeHighlighter.js b/node_modules/react-native-code-highlighter/dist/commonjs/lib/CodeHighlighter.js
index 0dfea90..0ef1ac7 100644
--- a/node_modules/react-native-code-highlighter/dist/commonjs/lib/CodeHighlighter.js
+++ b/node_modules/react-native-code-highlighter/dist/commonjs/lib/CodeHighlighter.js
@@ -17,6 +17,7 @@ const CodeHighlighter = ({
   textStyle,
   hljsStyle,
   scrollViewProps,
+  codeContainerStyle,
   containerStyle,
   ...rest
 }) => {
@@ -25,15 +26,16 @@ const CodeHighlighter = ({
     const classes = node.properties?.className ?? [];
     return classes.map(c => stylesheet[c]).filter(c => !!c);
   };
-  const renderNode = (nodes, keyPrefix = "row") => nodes.reduce((acc, node, index) => {
+  const renderNode = (nodes, keyPrefix = "row", depth = 0) => nodes.reduce((acc, node, index) => {
     const keyPrefixWithIndex = `${keyPrefix}_${index}`;
     if (node.children) {
       const styles = _reactNative.StyleSheet.flatten([textStyle, {
         color: stylesheet.hljs?.color
-      }, getStylesForNode(node)]);
+      }, depth === 0 ? codeContainerStyle : undefined, getStylesForNode(node)]);
       acc.push(/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.Text, {
         style: styles,
-        children: renderNode(node.children, `${keyPrefixWithIndex}_child`)
+        numberOfLines: depth === 0 ? 1 : undefined,
+        children: renderNode(node.children, `${keyPrefixWithIndex}_child`, depth + 1)
       }, keyPrefixWithIndex));
     }
     if (node.value) {
@@ -50,6 +52,7 @@ const CodeHighlighter = ({
       horizontal: true,
       contentContainerStyle: [stylesheet.hljs, scrollViewProps?.contentContainerStyle, containerStyle],
       children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
+        style: codeContainerStyle,
         onStartShouldSetResponder: () => true,
         children: renderNode(rows)
       })
diff --git a/node_modules/react-native-code-highlighter/dist/module/lib/CodeHighlighter.js b/node_modules/react-native-code-highlighter/dist/module/lib/CodeHighlighter.js
index 8eb424c..59f4d4c 100644
--- a/node_modules/react-native-code-highlighter/dist/module/lib/CodeHighlighter.js
+++ b/node_modules/react-native-code-highlighter/dist/module/lib/CodeHighlighter.js
@@ -11,6 +11,7 @@ export const CodeHighlighter = ({
   textStyle,
   hljsStyle,
   scrollViewProps,
+  codeContainerStyle,
   containerStyle,
   ...rest
 }) => {
@@ -19,15 +20,16 @@ export const CodeHighlighter = ({
     const classes = node.properties?.className ?? [];
     return classes.map(c => stylesheet[c]).filter(c => !!c);
   };
-  const renderNode = (nodes, keyPrefix = "row") => nodes.reduce((acc, node, index) => {
+  const renderNode = (nodes, keyPrefix = "row", depth = 0) => nodes.reduce((acc, node, index) => {
     const keyPrefixWithIndex = `${keyPrefix}_${index}`;
     if (node.children) {
       const styles = StyleSheet.flatten([textStyle, {
         color: stylesheet.hljs?.color
-      }, getStylesForNode(node)]);
+      }, depth === 0 ? codeContainerStyle : undefined, getStylesForNode(node)]);
       acc.push(/*#__PURE__*/_jsx(Text, {
         style: styles,
-        children: renderNode(node.children, `${keyPrefixWithIndex}_child`)
+        numberOfLines: depth === 0 ? 1 : undefined,
+        children: renderNode(node.children, `${keyPrefixWithIndex}_child`, depth + 1)
       }, keyPrefixWithIndex));
     }
     if (node.value) {
@@ -44,6 +46,7 @@ export const CodeHighlighter = ({
       horizontal: true,
       contentContainerStyle: [stylesheet.hljs, scrollViewProps?.contentContainerStyle, containerStyle],
       children: /*#__PURE__*/_jsx(View, {
+        style: codeContainerStyle,
         onStartShouldSetResponder: () => true,
         children: renderNode(rows)
       })
diff --git a/node_modules/react-native-code-highlighter/dist/typescript/lib/CodeHighlighter.d.ts b/node_modules/react-native-code-highlighter/dist/typescript/lib/CodeHighlighter.d.ts
index 9f0116e..3e3eaa3 100644
--- a/node_modules/react-native-code-highlighter/dist/typescript/lib/CodeHighlighter.d.ts
+++ b/node_modules/react-native-code-highlighter/dist/typescript/lib/CodeHighlighter.d.ts
@@ -6,6 +6,7 @@ export interface CodeHighlighterProps extends SyntaxHighlighterProps {
     hljsStyle: ReactStyle;
     textStyle?: StyleProp<TextStyle>;
     scrollViewProps?: ScrollViewProps;
+    codeContainerStyle?: StyleProp<ViewStyle>;
     /**
      * @deprecated Use scrollViewProps.contentContainerStyle instead
      */
